apollo-router-core/src/traits.rs

```diff
@@ -70,11 +70,11 @@ where
 
 impl<T: ?Sized> WithCaching for T where T: QueryPlanner + Sized + 'static {}
 
-/// An object that accepts a [`Request`] and allow creating [`PreparedQuery`]'s.
+/// An object that accepts a [`Request`] and allows creating [`PreparedQuery`]'s.
 ///
-/// The call to the function will either succeeds and return a [`PreparedQuery`] or it will fail and return
-/// a [`Response`] that can be returned immediately to the user. This is because GraphQL does
-/// not use the HTTP error codes, therefore it always return a response even if it fails.
+/// The call to the function will either succeeds and return a [`PreparedQuery`] or it will fail
```

Comment:
NIT: Might as well fix "succeeds" to "succeed" whilst we are fixing the other text.